### ISSUE TITLE
ERR_DLOPEN_FAILED on MacOS 10.14.6 using Node v16.13.1

### ISSUE THREAD

**Issue by raythree** [aws/aws-iot-device-sdk-js-v2#280]
### Describe the bug

Trying to follow the example to create a connection:

```
const endpoint = 'aXXXXXXXXXX-ats.iot.us-east-1.amazonaws.com';

let connection = null;

async function connect(clientId, certificate, privateKey) {
  const config_builder = iot.AwsIotMqttConnectionConfigBuilder.new_mtls_builder_from_path(certificate, privateKey);
  config_builder.with_clean_session(false);
  config_builder.with_client_id(clientId);
  config_builder.with_endpoint(endpoint);
  const config = config_builder.build();
  const client = new mqtt.MqttClient();
  connection = client.new_connection(config);
  
  log.info(`Connecting to ${endpoint}`);
  try {
    await connection.connect();
    log.info(`Connect OK`);
  }
  catch (err) {
    log.error(`Error connecting: ${err.message}`);
  }
}

connect("test-client", path-to-cert, path-to-key);
```
Getting a runtime error as shown below. Any suggestions welcome:

```
node:internal/modules/cjs/loader:1183
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: dlopen(/Users/bill/repos/cloud/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node, 1): no suitable image found.  Did find:
	/Users/bill/repos/hydra/cloud/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node: cannot load 'aws-crt-nodejs.node' (load command 0x80000034 is unknown)
	/Users/bill/repos/hydra/cloud/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node: cannot load 'aws-crt-nodejs.node' (load command 0x80000034 is unknown)
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1183:18)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/bill/repos/hydra/cloud/node_modules/aws-crt/dist/native/binding.js:59:19)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'ERR_DLOPEN_FAILED'
}

```

### SDK version used

"aws-iot-device-sdk-v2": "^1.8.7",

### Environment details (OS name and version, etc.)

MacOS 10.14.6, Node v16.13.1

**Comment by bretambrose** [comment]
What architecture is your Mac?  And just to verify, you have a simple sample project with that code and a single dependency to the v2 device sdk in package.json?

**Comment by raythree** [comment]
Hello, the system info:

```
  Model Name:	Mac Pro
  Model Identifier:	MacPro5,1
  Processor Name:	6-Core Intel Xeon
  Processor Speed:	3.33 GHz
  Number of Processors:	2
  Total Number of Cores:	12
  L2 Cache (per Core):	256 KB
  L3 Cache (per Processor):	12 MB
  Hyper-Threading Technology:	Enabled
  Memory:	32 GB
  Boot ROM Version:	140.0.0.0.0
  SMC Version (system):	1.39f5
  SMC Version (processor tray):	1.39f5
  Serial Number (system):	H09270S920H
  Serial Number (processor tray):	J5919015R1LUD     
  Hardware UUID:	C1D0F2C5-B2DE-5642-88AA-9A6D1E949F33
```

Yes, I have a simple sample project, code as above, with only the v2 SDK as dependency. Dev machine is a mac, target is Debian ... I have tried the V1 SDK on both, and it is fully functional, but I planned on starting with V2 as this is a new project. Thanks.

**Comment by raythree** [comment]
Sorry, what I mean is that in production, we will be running on Debian--that is our target for production, but all development is done on MacOS. I am seeing the problem with MacOS in development, running node 16 as reported, just running "npm install", then executing node to run the code above. However, since this seems to be an OS specific error, I ran with the V1 SDK on Debian and Mac, and both are fine.

**Comment by bretambrose** [comment]
It seems like there may be something odd with your install steps or directory structure:

/Users/bill/repos/cloud/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node

vs.

/Users/bill/repos/hydra/cloud/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node

in the output (additional hyrdra subdirectory).

Aws-crt-nodejs is a native extension and loading the .so/.dylib requires that file to be in an expected position at runtime.

Are you able to install and run the pub_sub sample in samples/node/pub_sub?

**Comment by raythree** [comment]
Yes, the sample I gave you did have an extra node_modules folder one level up,  as I had tried a test there. But it was the exact same package.json ... I created a new top level folder "/Users/bill/mqtt-test" and still see the same results. I will give the sample a try.

```
Error: dlopen(/Users/bill/mqtt-test/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node, 1): no suitable image found.  Did find:
	/Users/bill/mqtt-test/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node: cannot load 'aws-crt-nodejs.node' (load command 0x80000034 is unknown)
	/Users/bill/mqtt-test/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node: cannot load 'aws-crt-nodejs.node' (load command 0x80000034 is unknown)
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1183:18)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/bill/mqtt-test/node_modules/aws-crt/dist/native/binding.js:59:19)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'ERR_DLOPEN_FAILED'
```

**Comment by raythree** [comment]
I have the same issue with the pub-sub sample. Created a new folder in /Users/bill/mqtt-sample, using the contents of the sample:

```
[~/mqtt-sample] $ node dist/index.js
node:internal/modules/cjs/loader:1183
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: dlopen(/Users/bill/mqtt-sample/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node, 1): no suitable image found.  Did find:
	/Users/bill/mqtt-sample/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node: cannot load 'aws-crt-nodejs.node' (load command 0x80000034 is unknown)
	/Users/bill/mqtt-sample/node_modules/aws-crt/dist/bin/darwin-x64/aws-crt-nodejs.node: cannot load 'aws-crt-nodejs.node' (load command 0x80000034 is unknown)
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1183:18)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/bill/mqtt-sample/node_modules/aws-crt/dist/native/binding.js:59:19)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'ERR_DLOPEN_FAILED'
}

```

**Comment by raythree** [comment]
I am assuming it is something with this Mac, which is older. Tried the sample on a new Macbook pro, and it got farther than above--problem trying to require something from util (the sample folder is not stand alone and reaches into other parent folders). But it did not have the same ERR_DLOPEN_FAILED.

**Comment by xiazhvera** [comment]
For future references, the library loading issue should get resolved with version [v1.8.11](https://github.com/aws/aws-iot-device-sdk-js-v2/tree/v1.8.11). Please feel free to reopen the issue if you run into the problem again with v1.8.11 and a later version.
